src/core/stores/VNodeType.ts

```diff
@@ -0,0 +1,14 @@
+export enum VNodeType {
```

Comment:
Why not in `/typings` ?

Reply:
If find it strange to define types in differents ways:
- in the same file,
- in a separate file with the word type,
- in a separate file under the folder types,

Why not keep the types in the same file? It is directly related to VNode, it should reside in VNode.

Reply:
From the PR description:

> Note: I had to extract VNodeType out of VNode becauseVNode depended on Predicates to implement its method but Predicates depended on VNode because it needed VNodeType.

I am open to suggestions about how to solve this issue without exporting VNodeType in a different file.